### PR TITLE
Feature/add more endpoints and queryable submissions

### PR DESCRIPTION
**Pull request by otahontas**
### Description

This PR adds some new enpoints and makes submission now more easily queryable from db. Querying enpoint is still under construction. 

There are also little tweaks here and there (try-except checks etc.).

### Type of change
- [x] New feature (non-breaking change which adds functionality)

### Changes Made

Folowing endpoints are added:
```
- /objects/
    - GET: Fetches all objects available (study, sample, analysis... etc)
- /objects/<schema>/
    - POST: Adds this type of object to database
- /objects/<schema>/<accessionId>
    - GET: Fetches a single object with its accessionId
        - If extended with ?format=xml, returns originally submitted xml file corresponding this accessionId
- /<schema>/<accessionId>
   - GET: Alias for /objects/<schema>/<accessionId>
```
### Testing
- [x] Needs testing (start an issue or do a follow up PR about it)

### CODE REVIEW

File: metadata_backend/api/parser.py [hunk 1]
```diff
@@ -234,21 +211,47 @@ def parse_submission(self, data: Dict) -> Dict:
...
-        schema = self.load_schema(xml_type)
-        self.validate(content, schema)
-        content_json_raw = schema.to_dict(content)[xml_type.upper()][0]
-        return getattr(self, f"parse_{xml_type}")(content_json_raw)
+        order = {"study": 1, "sample": 2, "experiment": 3, "run": 4,
+                 "analysis": 5, "dac": 6, "policy": 7, "dataset": 8,
+                 "project": 9}
+        return sorted(data, key=lambda x: order[x["schema"]])
+
+    def _to_lowercase(self, obj):
```
Comment (blankdots) [inline metadata_backend/api/parser.py:240]: missing typing

File: metadata_backend/api/translator.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 """Handle translating API endpoint functionalities to CRUD operations."""
-
 from typing import Dict
 
 from aiohttp import web
+from bson import json_util
```
Comment (blankdots) [inline metadata_backend/api/translator.py:5]: more of a question: i see this is used just for `dumps` is there a difference between `json.dumps` ?

File: requirements.txt [hunk 3]
```diff
@@ -1,5 +1,5 @@
 aiohttp
-gunicorn
 pymongo
+python-dateutil
```
Comment (blankdots) [inline requirements.txt:3]: where is this used, should this be also in `setup.py` ?

why no more `gunicorn` ?

File: metadata_backend/api/translator.py (hunk 2 above)
Comment (otahontas) [reply metadata_backend/api/translator.py:5]: Yeah, json.dumps crashes when trying to serialize data returned from Mongo. More info here: https://api.mongodb.com/python/current/api/bson/json_util.html

Comment (blankdots) [reply metadata_backend/api/translator.py:5]: ok. was just wondering.

File: requirements.txt (hunk 3 above)
Comment (otahontas) [reply requirements.txt:3]: Dateutil was added for [adding publish dates for studies](https://github.com/CSCfi/metadata-submitter/blob/origin/add_more_endpoints_and_queryable_submissions/metadata_backend/api/parser.py#L97) (I think ENA uses +2 months from submission date by default). I also think some date-related parsing is needed later to handle those HOLD/SUPPRESS etc actions, so this library is probably useful.

I recall we thought that gunicorn is not needed. But anyways, it should be removed from [deploy/app.sh](https://github.com/CSCfi/metadata-submitter/blob/origin/add_more_endpoints_and_queryable_submissions/deploy/app.sh#L7) as well, if it it's not used. What do you think?

Comment (otahontas) [reply requirements.txt:3]: And setup.py gets parses requirements directly from requirements.txt: https://github.com/CSCfi/metadata-submitter/blob/master/setup.py#L5

Comment (blankdots) [reply requirements.txt:3]: ok to both :)
In `deploy/app.sh` we need to see. I would still prefer to have it there for the moment, but don't remember exactly if the s2i container installs it automatically or not.

**Pull request merged**